tests/common.cpp

```
@@ -42,6 +43,203 @@ namespace gdrcopy {
+            handle->allocated_size = allocated_size;
+
+            return CUDA_SUCCESS;
+        }
+
+        CUresult gpu_mem_free(gpu_mem_handle_t *handle)
+        {
+            CUresult ret = CUDA_SUCCESS;
+            CUdeviceptr ptr;
+
+            ret = cuMemFree(handle->unaligned_ptr);
+            if (ret == CUDA_SUCCESS)
+                memset(handle, 0, sizeof(gpu_mem_handle_t));
+
+            return ret;
+        }
+
+#if CUDA_VERSION >= 10020
+        CUresult gpu_vmm_alloc(gpu_mem_handle_t *handle, const size_t size, bool aligned_mapping, bool set_sync_memops)
+        {
+            CUresult ret = CUDA_SUCCESS;
+
+            size_t granularity, gran;
+            CUmemAllocationProp mprop;
+            CUdevice gpu_dev;
+            size_t rounded_size;
+            CUdeviceptr ptr = 0;
+            CUmemGenericAllocationHandle mem_handle = 0;
+            bool is_mapped = false;
+
+            #if CUDA_VERSION >= 11000
+            int RDMASupported = 0;
+            #endif
+
+            ret = cuCtxGetDevice(&gpu_dev);
+            if (ret != CUDA_SUCCESS) {
+                print_dbg("error in cuCtxGetDevice\n");
+                goto out;
+            }
+
+            #if CUDA_VERSION >= 11000
+            // CU_DEVICE_ATTRIBUTE_GPU_DIRECT_RDMA_WITH_CUDA_VMM_SUPPORTED is available from CUDA 11.0.
+            // For older CUDA versions, we will assume that VMM supports GDR.
+            // gdr_pin_buffer would return an error if it is not supported.
+            ret = cuDeviceGetAttribute(&RDMASupported, CU_DEVICE_ATTRIBUTE_GPU_DIRECT_RDMA_WITH_CUDA_VMM_SUPPORTED, gpu_dev);
+            if (ret != CUDA_SUCCESS) {
+                print_dbg("error in cuDeviceGetAttribute\n");
+                goto out;
+            }
+
+            if (!RDMASupported) {
+                print_dbg("GPUDirect RDMA is not supported on this GPU.\n");
+                ret = CUDA_ERROR_NOT_SUPPORTED;
+                goto out;
+            }
+            #endif
```

without GDR, this allocation is not really useful right?
so, we should return an error in the \#else branch.

---

Changed the version checking to CUDA 11.0 because VMM without RDMA is not meaningful anyway.